app/assets/javascripts/omni_nav/google-custom-search.js

```diff
@@ -0,0 +1,314 @@
+    };
+
+    var onSearchFilterClick = function() {
+      $selectedSearchFilter.text($(this).text());
+    };
+
+    var onSearchResultsResize = function() {
+      clearTimeout(resizeTimeoutId);
+      resizeTimeoutId = setTimeout(updateSearchResultsHeight, 500);
+    };
+
+    var bindAutocomplete = function() {
+      // Autocomplete map. This seems very fragile as it requires Google to use
+      // the same hardcorded selector each time. But it is how this was originally coded
+      // and so far has proved reliable so I am keeping it for now.
+      var contextSelectorMap = {
+        // omniNavContext: selector
+        'primary': 'table.gstl_50.gssb_c',
+        'utility': 'table.gstl_51.gssb_c'
+      };
+
+      // Bind only once: use a data flag to track binding.
+      var autoCompleteIsBound = $container.data('autoCompleteIsBound');
+      if ( autoCompleteIsBound ) return;
+
+      // This is required for selecting the Autocomplete block associated with
+      // SearchComponent. Since these are added post-loading by Google and are attached
+      // as absolute elements directly to the body, the parent cannot be deduced
+      // from the object itself. Checked Google API. No accessor. This is unfortunate.
+      var omniNavContext = ( containerId.indexOf("utility") >= 0 ) ? 'utility' : 'primary';
+      var autoCompleteSelector = contextSelectorMap[omniNavContext];
+      $container.data('autoCompleteIsBound', true);
+
+      // Bind callback: when autocomplete option is clicked, show results. Interval
+      // allows time for autoCompleteTable to be added to DOM.
+      var intervalId = setInterval(function() {
+        var $autoCompleteTable = $(autoCompleteSelector).find(".gsc-completion-container");
+        var autoCompleteLoaded = $autoCompleteTable.length > 0;
+
+        if ( autoCompleteLoaded ) {
+          $autoCompleteTable.on("click", onAutoCompleteClick);
+          clearInterval(intervalId);
+        }
+      }, 100);
+    };
+
+    var onAutoCompleteClick = function() {
+      setTimeout(showResults, 100);
+    };
+
+    var showResults = function() {
+      var term = gcsElement.getInputQuery();
+      var moreResultsUrl = SEARCH_RESULTS_BASE_URL + 'q=' + encodeURIComponent(term);
+      var hasSearchFilters = $container.find('.search-filter-option').length;
+
+      // Update More Results button.
+      $moreResultsButton.text('See more results for "' + term + '"');
+      $moreResultsButton.attr('href', moreResultsUrl);
+
+      // Mark results open.
+      $omniNav.addClass('search-results-open');
+      $container.addClass('search-results-open');
+
+      // Bind callbacks.
+      lockScroll();
+      $(document).on('keyup', onSearchEsc);
+      $(document).on('click', onDocumentClick);
+
+      // Trigger a click on the GCS tab corresponding to the search type dropdown selection
+      // unfortunately GCS does not provide a callback for when search results
+      // are returned so must use timeout
```

Comment:
A common theme with GCS. This was brought over from original code. Makes me wonder if we're not using GCS in best way, but it looks like neither Mirabel or I found a better way in docs.